DependencyInjection/HttplugExtension.php

```diff
@@ -137,6 +138,9 @@ private function configurePluginByName($name, Definition $definition, array $con
     {
         switch ($name) {
             case 'cache':
+                if (class_exists('Http\Client\Common\Plugin\CachePlugin')) {
```

Comment:
I would prefer a comment here that says why your are doing this. It seams kind of weird if you do not know that one namespace was deprecated and you are doing this because we want to keep BC. 

Same with the logger and stopwatch.